### ISSUE TITLE
heredoc unexpected is not a supported rule

### ISSUE THREAD

**Issue by issenn** [hadolint/hadolint#923]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

hadolint check pass.
load Dockerfile with `docker build` pass.

### Actual behavior

alert `hadolint: unexpected is not a supported rule` in vscode with hadolint
but load Dockerfile with `docker build` everything is fine.

```conslog
Dockerfile:74:1 unexpected 'E' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, a pragma, at least one space, or end of input
```

### Steps to reproduce the behavior

Add blow in a Dockerfile

this catch a error
```Dockerfile
RUN ls && \
    cat <<EOF >> go.mod && \
    cat go.mod
replace (
    github.com/user/repo => ../dir
)
EOF
```

This is ok, it works on first line, but I have more commands that need to be executed after.
```Dockerfile
RUN ls && cat <<EOF >> go.mod && \
    cat go.mod
replace (
    github.com/user/repo => ../dir
)
EOF
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
RUN ls && \
    cat <<EOF >> go.mod && \
    cat go.mod
replace (
    github.com/user/repo => ../dir
)
EOF
```

**Additional environment details (OS, stack version, etc.)**

**Comment by m-ildefons** [comment]
Hi Issenn,
Thanks for the bug report. I'm quite surprised this is even valid syntax. You are using buildkit, right?
You could use nested heredocs for the time being, until this is fixed:
```
# syntax=docker/dockerfile:1.3-labs
FROM debian:buster

RUN <<EOF
ls
cat <<MOD >> go.mod
replace (
  github.com/user/repo => ../dir
)
MOD
cat go.mod
# ... further commands here
EOF
```
